frameworks/react/src/AdvancedVideo.tsx

```diff
@@ -32,6 +33,10 @@ interface VideoProps {
   onEnded?: ReactEventHandler<any>
 }
 
+interface VideoProps extends BaseVideoProps {
+  innerRef?: ((instance: unknown) => void) | MutableRefObject<unknown> | null
```

Comment:
Why did we split VideoProps into two interfaces? 

Since innerRef is optional, is it possible to put it in the same interface?

Reply:
it's split because AdvancedVideoWithRef uses BaseVideoProps as props and translates the ref to innerRef.
We could do a simpler implementation (like it was done in cloudinary-react), where we don't allow the user to pass a ref like the react standard, but instead use needs to pass an 'innerRef' prop. In such an implementation we could use a single interface where innerRef is like you suggest.

Reply:
I think, relating to the the other comments as well, that it might be clearer to the user.

I think we need to find a better name than `innerRef` though to clearly express what this does.